catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -0,0 +1,302 @@
+    public UpdateResponse update(UpdateRequest updateRequest) throws IngestException {
+        return getProvider().update(updateRequest);
+    }
+
+    public void shutdown() {
+        closeSolrClient();
+    }
+
+    /**
+     * Shutdown the connection to Solr and releases resources.
+     */
+    private void closeSolrClient() {
+        LOGGER.debug("Closing connection to Solr client.");
+        if (getClient() != null) {
+            try {
+                getClient().close();
+            } catch (IOException e) {
+                LOGGER.info("Unable to close Solr client", e);
+            }
+        } else if (!clientFuture.isDone() && !clientFuture.isCancelled()) {
+            clientFuture.cancel(true);
+        }
+        LOGGER.debug("Finished closing connection to Solr client.");
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
+        updateClient(PropertyResolver.resolveProperties(url));
+    }
+
+    /**
+     * Updates the configuration of Solr if necessary
+     *
+     * @param urlValue - url to Solr
+     */
+    public void updateClient(String urlValue) {
+        LOGGER.debug("New url {}", urlValue);
+
+        if (urlValue != null) {
+            if (!StringUtils.equalsIgnoreCase(urlValue.trim(), url) || getClient() == null) {
+                url = urlValue.trim();
+
+                if (getClient() != null) {
+                    LOGGER.debug(
+                            "Shutting down the connection manager to Solr and releasing allocated resources.");
+                    closeSolrClient();
+                    LOGGER.debug("Shutdown complete.");
+                }
+
+                updateClient();
+            }
+
+        } else {
+            url = null;
+        }
+    }
+
+    public void updateClient() {
+        clientFuture = createClient();
+        client = null;
+    }
+
+    protected abstract Future<SolrClient> createClient();
+
+    private CatalogProvider getProvider() {
+
+        if (!isClientConnected(getClient())) {
+            return new UnavailableSolrCatalogProvider();
+        }
+
+        if (provider instanceof UnavailableSolrCatalogProvider) {
+            provider = new SolrCatalogProvider(getClient(),
+                    filterAdapter,
+                    solrFilterDelegateFactory,
+                    resolver);
+        }
+
+        provider.maskId(getId());
+        return provider;
+
+    }
+
+    private boolean isClientConnected(SolrClient solr) {
+
+        if (solr == null) {
+            return false;
+        }
+
+        try {
+            return OK_STATUS.equals(solr.ping()
+                    .getResponse()
+                    .get("status"));
+        } catch (Exception e) {
+            /*
+             * if we get any type of exception, whether declared by Solr or not, we do not want to
+             * fail, we just want to return false
+             */
+            LOGGER.info(PING_ERROR_MESSAGE, e);
```

Comment:
I moved the stack trace down to debug.  I left the ping message at info since the code using the Solr client will log or send an error back to the user if the ping fails.